api/base/utils.py

```diff
@@ -22,3 +27,31 @@ def get_object_or_404(model_cls, query_or_pk):
         return model_cls.find_one(query)
     except NoResultsFound:
         raise NotFound
+
+
+def waterbutler_url_for(request_type, provider, path, node_id, token, obj_args=None, **query):
```

Comment:
Would it be possible to define the function in this module but then import it in `website.util`, so it can be imported from there as well?